### DISCUSSION TITLE
Nested Routes based on GrahpQL Data

### DISCUSSION THREAD

**Discussion by Yofret-Rios** [gatsbyjs/gatsby#37755] [Help] [answered]
Hi Everyone, I'm currently trying to generate an URL structure like this one:
```
/region/greece
/region/hungary
...
/specialty/eco-friendly
/specialty/light
/specialty/full
...
```
Using the following data from graphQL (already stored)

```
{
  "data": {
    "allCategoryPath": {
      "nodes": [
        {
          "name": "region",
          "subpath": [
            {
              "name": "greece"
            },
            {
              "name": "hungary"
            }
          ]
        },
        {
          "name": "specialty",
          "subpath": [
            {
              "name": "eco-friendly"
            },
            {
              "name": "light"
            },
            {
              "name": "full"
            }
          ]
        }
      ]
    }
  },
  "extensions": {}
}
```
I'm using the File System API documentation but it doesn't seem to be working:
{CategoryPath.name}/{CategoryPath.subpath__name}.js

```
- {CategoryPath.name}
-- {CategoryPath.subpath__name}.js
```

Gatsby throws the error shown in the screenshot, I'm pretty sure that it is possible, but I might be missing something, Thanks in advance for any provided help!

<img width="1166" alt="Screenshot 2023-03-16 at 6 50 04 PM" src="https://user-images.githubusercontent.com/81445372/225945596-06903daa-39a8-4b5b-8b29-b05603e3b3bc.png">

**Comment by LekoArts** [comment, marked as answer]
Hi!

Unfortunately you hit an edge-case that is not supported (see comment https://github.com/gatsbyjs/gatsby/discussions/26375?sort=top#discussioncomment-2576360).
Ideally this information should be in a "known limitations" section at https://www.gatsbyjs.com/docs/reference/routing/file-system-route-api/

I'd suggest using the createPages API in `gatsby-node` which will enable you to do this.

**Comment by Yofret-Rios** [comment]
Hi @LekoArts, thanks for the recommendations, I've changed my approach to a programmatically, yes I do agree that it should be in the known limitations part, I'm marking it as resolved
